plugconf/plugconf.go

```diff
@@ -519,6 +543,34 @@ func getDependencies(fn *ast.Function) (pathutil.ReposPathList, error) {
 	return deps, parseErr
 }
 
+// shouldLoadUnconditionally checks if the given function only contains the
+// single default 'return 1' statement which means loading unconditionally.
+// If any other statement is found it is considered a condition.
+func shouldLoadUnconditionally(fn *ast.Function) bool {
```

Comment:
Could this function's return type be true, false, or "cannot determinated" ?
Can determine literally not to load plugconf when `val.Value` is `0`, `v:false`.

Reply:
so I can add more literal values later.

Reply:
What exactly do you have in mind?

Return a `*bool` which can be `nil` if it "cannot be determined"? That seems to be a bit like a misuse of the datatype.

Or define an enum instead:

```
type LoadCondition int

const (
	ShouldNotLoad = iota
	ShouldLoad
	Undefined
)

func shouldLoadUnconditionally(fn *ast.Function) LoadCondition {
```

While this theoretically can be extended I cannot image another valid value to be returned from a function "shouldLoadUnconditionally".

But maybe I have misunderstood you here.

Reply:
I prefer enum.

> While this theoretically can be extended I cannot image another valid value to be returned from a function "shouldLoadUnconditionally".

I imagined static analysis for other types like String, None, ...
but user doesn't write the code to return such values in `s:should_load()`.
indeed it is not necessary to be extendable function, I didn't think deeply, sorry.

Reply:
OK, so do you still want to me to change anything?

Reply:
yes, please change to enum way.
I also want to detect `ShouldNotLoad` literally.

Reply:
I am still not sure I understood you correctly. What do you mean by

> I also want to detect ShouldNotLoad literally.

?

To be clear about the purpose of the `shouldLoadUncoditionally` method: This method only checks whether the `s:shouldLoad()` function in the plugconf has changed from its default value or not. If it is unchanged, that means to "load unconditionally" and there is no need to actually copy that function to the target `.vim/pack/volt/start/system/plugin/bundled_plugconf.vim` and execute it. 

In all other cases (that means the user actually made some changes to `s:shouldLoad` function) this function will be included in the `bundled_plugconf.vim` and executed before loading the plugin.

Therefore I don't even see a value in the enum, since the option `Undefined` will never match.

Reply:
sorry for late reply. I misunderstood the behavior.
it is okay not to fix it.

I thought `s:should_load()` is recognized at runtime (by Vim).
but actually it is recognized statically/literally (by Volt).